### PR TITLE
Adjust last_reset calculation to handle charge_energy_added being the same

### LINKED ISSUE #288: `last_reset` is changed every time the integration updates
```
31976|3172176861|{"state_class":"total_increasing","charger_phases":null,"charge_limit_soc":90,"last_reset":"2022-03-19 00:45:47.002129","unit_of_measurement":"kWh","device_class":"energy","friendly_name":"Model X'y energy added sensor"}
31977|2391666629|{"state_class":"total_increasing","charger_phases":null,"charge_limit_soc":90,"last_reset":"2022-03-19 00:45:57.002270","unit_of_measurement":"kWh","device_class":"energy","friendly_name":"Model X'y energy added sensor"}
31978|2731341971|{"state_class":"total_increasing","charger_phases":null,"charge_limit_soc":90,"last_reset":"2022-03-19 00:46:07.001811","unit_of_measurement":"kWh","device_class":"energy","friendly_name":"Model X'y energy added sensor"}
31979|2927162728|{"state_class":"total_increasing","charger_phases":null,"charge_limit_soc":90,"last_reset":"2022-03-19 00:46:17.001752","unit_of_measurement":"kWh","device_class":"energy","friendly_name":"Model X'y energy added sensor"}
```
Screen Shot 2022-03-18 at 14 49 00
Screen Shot 2022-03-18 at 14 48 55

### PR DESCRIPTION
**Pull request by bdraco**
If the value was the same as the last update last_reset would be set

Fixes #288

### CODE REVIEW

**Comment by bdraco** [comment]
Well on restart its now

`317314|sensor.xizzy_energy_added_sensor|4.42||328178|2022-03-24 08:08:51.004084|2022-03-24 08:08:51.004084|316862|1380|1380|3055519725|{"state_class":"total_increasing","charger_phases":null,"charge_limit_soc":90,"last_reset":"00:00:00","unit_of_measurement":"kWh","device_class":"energy","friendly_name":"XIZZY energy added sensor"}` because the condition doesn't hit

File: teslajsonpy/homeassistant/charger.py [hunk 1]
```diff
@@ -360,7 +360,7 @@ def refresh(self) -> None:
             self.__charge_limit_soc = data["charge_limit_soc"]
             self.__charger_power = data["charger_power"]
             self.attrs["charge_limit_soc"] = self.charge_limit_soc
-            self.attrs["last_reset"] = str(self.last_reset)
+            self.attrs["last_reset"] = self.last_reset
```
Comment (bdraco) [inline teslajsonpy/homeassistant/charger.py:363]: https://developers.home-assistant.io/docs/core/entity/sensor?_highlight=last_reset#properties

It looks like this is supposed to be a `datetime.datetime`

**Comment by bdraco** [comment]
AFAICT its working now

**Comment by alandtse** [comment]
Thanks for catching that issue. I believe I added the str cast because at one point there were errors having a `datetime.datetime` or `None` as an attribute (I vaguely recall a logging error). I assume you're not seeing any errors now?

**Comment by bdraco** [comment]
All working for me.  No errors

**Pull request merged**